addons/website_sale/models/website.py

```diff
@@ -213,6 +213,7 @@ def _prepare_sale_order_values(self, partner, pricelist):
         company = self.company_id or pricelist.company_id
         if company:
             values['company_id'] = company.id
+            values['note'] = self.env['ir.config_parameter'].sudo().get_param('sale.use_sale_note') and company.sale_note or ""
```

Comment:
I think this is the most logical behavior:
```suggestion
            if self.env['ir.config_parameter'].sudo().get_param('sale.use_sale_note'):
                values['note'] = company.sale_note or ""
```

You fill the `note` field only if you need default T&C, otherwise you keep the default behavior. What do you think?

Reply:
not at all, your solution looks more logical in my opinion as well.
I did update the code as you suggested.